addons/web/static/src/core/domain_selector/domain_selector_autocomplete.js

```diff
@@ -0,0 +1,182 @@
+        return props.value.filter((val) => this.isId(val));
+    }
+
+    getTags(props, displayNames) {
+        return props.value.map((val, index) => {
+            let text;
+            let colorIndex;
+            if (this.isId(val)) {
+                text =
+                    displayNames[val] !== null
+                        ? displayNames[val]
+                        : _t(`Missing record ID: ${val}`);
+                colorIndex = displayNames[val] !== null ? 0 : 2; // 0 = grey, 2 = orange
+            } else {
+                text =
+                    val instanceof DomainValueExpr
+                        ? val.expr
+                        : _t(`Invalid record ID: ${formatAST(toPyValue(val))}`);
+                colorIndex = val instanceof DomainValueExpr ? 2 : 1; // 1 = red
+            }
+            return {
+                text,
+                colorIndex,
+                onDelete: () => {
+                    this.props.update([
+                        ...this.props.value.slice(0, index),
+                        ...this.props.value.slice(index + 1),
+                    ]);
+                },
+            };
+        });
+    }
+
+    isId(val) {
+        return Number.isInteger(val) && val >= 1;
+    }
+
+    async loadOptionsSource(name) {
+        if (this.lastProm) {
+            this.lastProm.abort(false);
+        }
+        this.lastProm = this.search(name, SEARCH_LIMIT + 1);
+        const nameGets = await this.lastProm;
+        this.addNames(nameGets);
+        const options = nameGets.map(([value, label]) => ({ value, label: label.split("\n")[0] }));
+        if (SEARCH_LIMIT < nameGets.length) {
+            options.push({
+                label: this.env._t("Search More..."),
+                action: this.onSearchMore.bind(this, name),
+                classList: "o_m2o_dropdown_option",
+            });
+        }
+        if (options.length === 0) {
+            options.push({ label: _t("(no result)"), unselectable: true });
+        }
+        return options;
+    }
+
+    async onSearchMore(name) {
+        const { fieldString, resModel } = this.props;
+        let operator;
+        const ids = [];
+        if (name) {
+            const nameGets = await this.search(name, SEARCH_LIMIT);
+            this.addNames(nameGets);
+            operator = "in";
+            ids.push(...nameGets.map((nameGet) => nameGet[0]));
+        } else {
+            operator = "not in";
+            ids.push(...this.getIds());
+        }
+        const dynamicFilters = ids.length
+            ? [
+                  {
+                      description: sprintf(this.env._t("Quick search: %s"), name),
+                      domain: [["id", operator, ids]],
+                  },
+              ]
+            : undefined;
+        // fine for now but we don't like this kind of dependence of core to views
+        const SelectCreateDialog = registry.category("dialogs").get("select_create");
```

Comment:
Use a simple import of the dialog in DomainSelectorAutocomplete.
It's maybe not great but its still better than adding a new registry.